contrib/compile-tests.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+#
+# Checks intended compile failures.
+
+set -euo pipefail
+
+REPO_DIR=$(git rev-parse --show-toplevel)
+
+should_pass() {
+    result=0
+    # All crates in the subdirectory pass/ must compile
+    for dir in pass/*; do
+        if [ -d "$dir" ]; then
+            pushd "$dir" > /dev/null
+            output=$(cargo build 2>&1)
+            if [ $? -ne 0 ]; then
+                echo "$output"
+                echo "compile-tests/$dir/ failed to compile"
+                result=1
+            fi
+            popd > /dev/null
+        fi
+    done
+    return "$result"
+}
+
+should_fail() {
+    result=0
+    # All crates in the subdirectory fail/ must fail to compile
+    for dir in fail/*; do
+        if [ -d "$dir" ]; then
+            pushd "$dir" > /dev/null
+            output=$(cargo build 2>&1)
+            if [ $? -eq 0 ]; then
+                echo "$output"
```

Comment:
If compilation succeeded and it shouldn't there will be no useful information printed by `cargo`. I'd rather get rid of it.

Reply:
I have removed the cargo output for this case